0.1.0/lib.typ

```diff
@@ -0,0 +1,183 @@
+#import "@preview/acrostiche:0.3.1": init-acronyms, print-index
+#import "lib/research-questions.typ": init-rqs
+#import "lib/utils.typ": current-academic-year
+
+#let _todo-col = rgb("#ff5a08")
+#let _idea-col = rgb("#2b8a70")
+
+#let thesis(
+  // The title of this thesis [content]
+  title: none,
+  // the authors of this thesis [array of strings]
+  authors: none,
+  // the font that's used for the thesis [string]
+  font: "UGent Panno Text",
+  // optionally align pagebreaks to odd pages [bool]
+  odd_pagebreaks: false,
+  // the actual content of the thesis
+  body
+) = {
+  // title, authors are required (return clear error message if not given)
+  assert.ne(title, none, message: "`title` is a required argument")
+  assert.ne(authors, none, message: "`authors` is a required argument")
+
+  set document(
+    title: title, 
+    author: authors,
+  )
+
+  show par: set block(spacing: 20pt)
+  set par(leading: 12pt, justify: true)
+  set page(
+    margin: (left: 2.5cm, right: 2.5cm, top: 2.5cm, bottom: 2.5cm),  
+    paper: "a4",
+    numbering: "I",
+    header: context {
+      let elems = query(
+        selector(heading).before(here()))
+
+      let headings_at_this_page = query(
+        heading.where(level: 1)
+      ).find(h => h.location().page() == here().page())
+
+      let page_has_no_heading = headings_at_this_page == none 
+
+      if elems.len() != 0 and page_has_no_heading {
+        let body = elems.last().body
+        align(right, emph(body))
+      }
+    },
+  )
+  
+  
+  show heading.where(
+    level: 2
+  ): it => {
+    text(size: 26pt, it)
+    v(1em)
+  }
+  
+  show heading.where(
+    level: 3
+  ): it => {
+    text(size: 20pt, it)
+    v(.75em)
+  }
+  
+  set text(font: font)
+  // don't break up words in justified text
+  set text(hyphenate: false)
+  
+  // make new sections appear on the right hand side
+  if odd_pagebreaks {
+    set pagebreak(weak: true, to: "odd")
```

Comment:
how about we add an argument `pagebreak-to`, have it default `"odd"` and pass that to `to`. Or is that too much indirection?

Reply:
Will anyone ever set it to something else than "odd" or not setting it at all?
If there's just these two scenarios, a bool is more clear I think

Reply:
I kept it as a boolean for now, feel free to change later